Adding function to find lessons containing specific terms [WIP]

ian-flores opened the pull request:
Hi @carpentries/glosario-admins , 

I've added the `find_lessons()` function which finds lessons containing specific terms to be looked for. It allows for a flexible folder structure, but also for providing the name of the individual files if the end-user prefers so. Eager to hear your thoughts on this.

zkamvar reviewed (commented):
I think it's a good start and it's definitely a good idea to give users an idea of what is defined in their compendium.

I have a few technical comments below, but I will also suggest to reconsider the name of the function. When I think `find_lessons()`, I am thinking of a function that will find my lesson materials for me, not a function that will display a list of files with their associated function definitions. Maybe `find_definitions()` or `defined()`?

ian-flores commented on R/find_lessons.R:
```diff
@@ -0,0 +1,52 @@
+  terms_returned <- list()
+
+  # Iterate over the YAML headers to search for the terms
+  for (idx in 1:length(yaml_headers)){
```
I didn't knew this one. Thanks!

ian-flores commented:
Hi @zkamvar I incorporated the changes you mentioned above. I also renamed the function and the docs to `find_terms` instead of `find_lessons` so that it better fits the purpose of the function. Thanks for taking the time to do this review!
